### PR TITLE
chore(environment configuration): tools cz, workflows, hooks, codeowners, venv

### PR DESCRIPTION
**Pull request by nezno4**
# chore(environment configuration): tools cz, workflows, hooks, codeowners, venv
--

## What? ✌️

The following tools have been configured in the project:

- [Commitizen](https://commitizen-tools.github.io/commitizen/),
- [DeepSource](https://deepsource.com/),
- .gitignore,
- [README.md](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax),
- PULL_REQUEST_TEMPLATE.md --> [1](https://contribute.freecodecamp.org/how-to-open-a-pull-request/) [2](https://www.pullrequest.com/blog/writing-a-great-pull-request-description/),
- requirements.txt
- CODEOWNERS
- hook:
   - pre-commit 
- workflows:
  - ci.yml ---> responsible for automates code quality checks, testing, and version management
- virtual environment for python [3.13.1] .venv

## What Implement successful? 🤔

- [x] .cz.toml
- [x] .deepsource.toml
- [x] .gitignore
- [x] requirements.txt
- [x] README.md
- [x] PULL_REQUEST_TEMPLATE.md
- [x] CODEOWNERS
- [x] \workflows\ci.yml
- [x] .pre-commit-config.yaml
- [x] venv

issue: -

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,269 @@
...
+    G --> H[Run Pre-commit Hooks]
+    H --> I[Lint Code - flake8, black]
+    I --> J[Run Tests with Coverage]
+    J --> K[Run DeepSource Analysis]
+
+    K --> L{Is on main branch?}
+    L -->|Yes| M[Bump Version and Create Tag]
+    L -->|No| N[End]
+	
+	M --> O[Generate CHANGELOG.md]
```
Comment (Nezno6) [inline README.md:266]: remove indent

**Review by Nezno6 — DISMISSED** [review]
it is ok

File: README.md (hunk 1 above)
Comment (nezno4) [reply README.md:266]: ok

**Review by Nezno6 — APPROVED** [review]
It is ok

**Pull request merged**